src/ServerRequestCreator.php

```diff
@@ -51,13 +51,30 @@ public function fromGlobals(): ServerRequestInterface
 
         $headers = \function_exists('getallheaders') ? getallheaders() : static::getHeadersFromServer($_SERVER);
 
-        return $this->fromArrays($server, $headers, $_COOKIE, $_GET, $_POST, $_FILES, \fopen('php://input', 'r') ?: null);
+        $post = null;
+        if ('POST' === $this->getMethodFromEnv($server)) {
+            foreach ($headers as $headerName => $headerValue) {
+                if ('content-type' !== \strtolower($headerName)) {
+                    continue;
+                }
+                if (\in_array(
+                    \strtolower(\trim(\explode(';', $headerValue, 2)[0])),
+                    ['application/x-www-form-urlencoded', 'multipart/form-data']
+                )) {
+                    $post = $_POST;
```

Comment:
What will $_POST contain if it isnt a POST request with content type `application/x-www-form-urlencoded or multipart/form-data`?

Reply:
It is just an empty array.

Reply:
Yes. From my reading of PSR-7, `getParsedBody` should return `null` when no (parseable) request body was provided. If we pass along the empty array from `$_POST` that would be the same as saying we successfully parsed the body content, which may not have been try at al. E.g. we make no attempt to parse JSON bodies.

That is why I wanted to establish a clear difference between `[]` and `null` 😄

Reply:
Thank you